packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -636,6 +644,79 @@ export class EthereumConnection implements BcpConnection {
     return { ...transaction, fee: await this.getFeeQuote(transaction) };
   }
 
+  public async getSwaps(query: AtomicSwapQuery): Promise<ReadonlyArray<AtomicSwap>> {
+    if (isAtomicSwapIdQuery(query)) {
+      const data = Uint8Array.from([...Abi.calculateMethodId("get(bytes32)"), ...query.swapid]);
+
+      const params = [
+        {
+          to: constants.atomicSwapEtherContractAddress,
+          data: "0x" + Encoding.toHex(data),
+        },
+      ] as ReadonlyArray<any>;
+      const swapsResponse = await this.rpcClient.run({
+        jsonrpc: "2.0",
+        method: "eth_call",
+        params: params,
+        id: 7,
+      });
+      if (isJsonRpcErrorResponse(swapsResponse)) {
+        throw new Error(JSON.stringify(swapsResponse.error));
+      }
+
+      if (swapsResponse.result === null) {
+        return [];
+      }
+
+      const senderBegin = 12;
+      const senderEnd = senderBegin + 20;
```

Comment:
I think it is easier to understand the code if we slice them field by field. In this case 4 bytes method ID, than 32 bytes sender (zero padded). The address can be extracted from a 32 bytes field using Abi.decodeAddress.

Reply:
Sounds good but there's no method ID in the return value.

Reply:
Yeah right. The number 12 is so unusual in computer science that I was confused.